<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Lobby</class>
 <widget class="QMainWindow" name="Lobby">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1241</width>
    <height>734</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Need for Speed - Lobby</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">QPushButton {
    background-color: #ff6600;   /* color base */
    color: white;
    font-weight: bold;
    border: none;
    border-radius: 10px;         /* podés ajustar el radio si querés redondeado */
    padding: 8px;
}

QPushButton:hover {
    background-color: #ff8800;   /* color al pasar el mouse */
}
</string>
      </property>
      <property name="currentIndex">
       <number>8</number>
      </property>
      <widget class="QWidget" name="page_connection">
       <widget class="QPushButton" name="connectButton">
        <property name="geometry">
         <rect>
          <x>520</x>
          <y>390</y>
          <width>171</width>
          <height>51</height>
         </rect>
        </property>
        <property name="text">
         <string>Connect</string>
        </property>
       </widget>
       <widget class="QLabel" name="game_name">
        <property name="geometry">
         <rect>
          <x>120</x>
          <y>150</y>
          <width>981</width>
          <height>131</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">QLabel#game_name {
    color: #1cd3dc; /* Relleno azul celeste */
    font-family: 'Tektur';
    font-size: 90px;
    font-weight: 900;
    qproperty-alignment: 'AlignCenter';
    text-transform: uppercase;
    letter-spacing: 6px;
    padding: 25px 60px;
    border-radius: 25px;

    background-color: rgba(5, 51, 86, 0.85);
    border: 4px solid #1cd3dc;
    box-shadow: 0px 0px 20px rgba(28, 211, 220, 0.5);

    /* Contorno negro &quot;falso&quot; con varias sombras alrededor del texto */
    text-shadow:
        -2px  0px 0 #000,
         2px  0px 0 #000,
         0px  2px 0 #000,
         0px -2px 0 #000,
        -2px -2px 0 #000,
         2px -2px 0 #000,
        -2px  2px 0 #000,
         2px  2px 0 #000;
}
</string>
        </property>
        <property name="text">
         <string>NEED FOR SPEED</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QLabel" name="flag_left">
        <property name="geometry">
         <rect>
          <x>-70</x>
          <y>170</y>
          <width>191</width>
          <height>101</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">QLabel#flag_left,
QLabel#flag_right {
    background-image: url(:/flag_pattern.png);
    background-repeat: repeat-x;
    background-position: center;
    background-size: 50px 50px;   /* Escala del patrón: más chico = más cuadros visibles */
    border: none;
    min-width: 150px;
    min-height: 40px;
}

</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QLabel" name="flag_right">
        <property name="geometry">
         <rect>
          <x>1100</x>
          <y>170</y>
          <width>181</width>
          <height>101</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">QLabel#flag_left,
QLabel#flag_right {
    background-image: url(:/flag_pattern.png);
    background-repeat: repeat-x;
    background-position: center;
    background-size: 50px 50px;   /* Escala del patrón: más chico = más cuadros visibles */
    border: none;
    min-width: 150px;
    min-height: 40px;
}

</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="page_username">
       <widget class="QFrame" name="backgorund">
        <property name="geometry">
         <rect>
          <x>350</x>
          <y>189</y>
          <width>500</width>
          <height>321</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">
    QFrame {
        background-color: rgba(0, 0, 0, 0.5);
        border: 2px solid #ff6600;
        border-radius: 20px;
    }
    QLabel {
        color: white;
        font-family: 'Tektur';
        font-size: 20px;
        font-weight: bold;
    }
    QLineEdit {
        background: rgba(255,255,255,0.1);
        color: white;
        border: 2px solid #ff6600;
        border-radius: 10px;
        padding: 8px;
        font-family: 'Tektur';
        font-size: 17px;
    }
    QLineEdit::placeholder {
        color: rgba(255,255,255,180);
        font-style: italic;
    }
    QPushButton {
        background-color: #ff6600;
        color: white;
        border: none;
        border-radius: 10px;
        padding: 10px 20px;
        font-family: 'Tektur';
        font-size: 16px;
        font-weight: bold;
    }
    QPushButton:hover { background-color: #ffa31a; }
    QPushButton:pressed { background-color: #cc5200; }
         </string>
        </property>
        <widget class="QLineEdit" name="input_username">
         <property name="geometry">
          <rect>
           <x>90</x>
           <y>90</y>
           <width>321</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="placeholderText">
          <string>Enter your username</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnConfirmUsername">
         <property name="geometry">
          <rect>
           <x>140</x>
           <y>190</y>
           <width>211</width>
           <height>45</height>
          </rect>
         </property>
         <property name="text">
          <string>Confirm username</string>
         </property>
        </widget>
       </widget>
      </widget>
      <widget class="QWidget" name="page_rooms">
       <property name="styleSheet">
        <string notr="true">
QWidget {
    background-color: rgba(0, 0, 0, 0.5);
    border-radius: 10px;
}
QLabel {
    color: white;
    font-family: 'Tektur';
    font-size: 22px;
    font-weight: bold;
}
QListWidget {
    background-color: rgba(255,255,255,0.1);
    border-radius: 10px;
    color: white;
    font-family: 'Tektur';
    font-size: 16px;
}
QPushButton {
    background-color: #ff6600;
    color: white;
    border: none;
    border-radius: 10px;
    padding: 10px 20px;
    font-family: 'Tektur';
    font-size: 16px;
    font-weight: bold;
}
QPushButton:hover { background-color: #ffa31a; }
QPushButton:pressed { background-color: #cc5200; }
  </string>
       </property>
       <widget class="QLabel" name="labelRoomsTitle">
        <property name="geometry">
         <rect>
          <x>470</x>
          <y>100</y>
          <width>300</width>
          <height>40</height>
         </rect>
        </property>
        <property name="text">
         <string>Available Rooms</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QListWidget" name="listRooms">
        <property name="geometry">
         <rect>
          <x>370</x>
          <y>160</y>
          <width>500</width>
          <height>320</height>
         </rect>
        </property>
       </widget>
       <widget class="QPushButton" name="btnPrevRooms">
        <property name="geometry">
         <rect>
          <x>410</x>
          <y>500</y>
          <width>120</width>
          <height>40</height>
         </rect>
        </property>
        <property name="text">
         <string>&lt; Prev</string>
        </property>
       </widget>
       <widget class="QLabel" name="labelPageInfo">
        <property name="geometry">
         <rect>
          <x>560</x>
          <y>500</y>
          <width>120</width>
          <height>40</height>
         </rect>
        </property>
        <property name="text">
         <string>Page 1 / 1</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QPushButton" name="btnNextRooms">
        <property name="geometry">
         <rect>
          <x>710</x>
          <y>500</y>
          <width>120</width>
          <height>40</height>
         </rect>
        </property>
        <property name="text">
         <string>Next &gt;</string>
        </property>
       </widget>
       <widget class="QPushButton" name="btnContinueRooms">
        <property name="geometry">
         <rect>
          <x>530</x>
          <y>560</y>
          <width>180</width>
          <height>45</height>
         </rect>
        </property>
        <property name="text">
         <string>Continue</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="page_menu">
       <widget class="QPushButton" name="btnCreate">
        <property name="geometry">
         <rect>
          <x>480</x>
          <y>260</y>
          <width>251</width>
          <height>61</height>
         </rect>
        </property>
        <property name="text">
         <string>Create Game</string>
        </property>
       </widget>
       <widget class="QPushButton" name="btnJoin">
        <property name="geometry">
         <rect>
          <x>480</x>
          <y>360</y>
          <width>251</width>
          <height>61</height>
         </rect>
        </property>
        <property name="text">
         <string>Join Game</string>
        </property>
       </widget>
       <widget class="QPushButton" name="btnMap">
        <property name="geometry">
         <rect>
          <x>480</x>
          <y>450</y>
          <width>251</width>
          <height>61</height>
         </rect>
        </property>
        <property name="text">
         <string>Create race tracks</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="page_create">
       <property name="styleSheet">
        <string notr="true">
         QWidget {
            background-color: rgba(0, 0, 0, 0.5);
            border-radius: 10px;
         }
         QLabel {
            color: white;
            font-family: 'Tektur';
            font-size: 18px;
            font-weight: bold;
         }
         QPushButton {
            background-color: #ff6600;
            color: white;
            border: none;
            border-radius: 10px;
            padding: 10px 20px;
            font-family: 'Tektur';
            font-size: 16px;
            font-weight: bold;
         }
         QPushButton:hover { background-color: #ffa31a; }
         QPushButton:pressed { background-color: #cc5200; }
        </string>
       </property>
       <widget class="QLabel" name="labelCode">
        <property name="geometry">
         <rect>
          <x>490</x>
          <y>150</y>
          <width>250</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string>Room code:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="margin">
         <number>-2</number>
        </property>
       </widget>
       <widget class="QLabel" name="labelCreateInfo">
        <property name="geometry">
         <rect>
          <x>310</x>
          <y>219</y>
          <width>611</width>
          <height>51</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QPushButton" name="btnContinue">
        <property name="geometry">
         <rect>
          <x>520</x>
          <y>450</y>
          <width>180</width>
          <height>45</height>
         </rect>
        </property>
        <property name="text">
         <string>Continue</string>
        </property>
       </widget>
       <widget class="QComboBox" name="comboMaxPlayers">
        <property name="geometry">
         <rect>
          <x>470</x>
          <y>360</y>
          <width>291</width>
          <height>41</height>
         </rect>
        </property>
        <property name="maxVisibleItems">
         <number>6</number>
        </property>
        <item>
         <property name="text">
          <string/>
         </property>
        </item>
        <item>
         <property name="text">
          <string>2</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>3</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>4</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>5</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>6</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>7</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>8</string>
         </property>
        </item>
       </widget>
       <widget class="QLabel" name="labelPlayersTitle">
        <property name="geometry">
         <rect>
          <x>490</x>
          <y>300</y>
          <width>250</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string>How many players?</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="page_join">
       <widget class="QFrame" name="frameJoin">
        <property name="geometry">
         <rect>
          <x>370</x>
          <y>120</y>
          <width>500</width>
          <height>400</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">
    QFrame {
        background-color: rgba(0, 0, 0, 0.5);
        border: 2px solid #ff6600;
        border-radius: 20px;
    }
    QLabel {
        color: white;
        font-family: 'Tektur';
        font-size: 20px;
        font-weight: bold;
    }
    QLineEdit {
        background: rgba(255,255,255,0.1);
        color: white;
        border: 2px solid #ff6600;
        border-radius: 10px;
        padding: 8px;
        font-family: 'Tektur';
        font-size: 17px;
    }
    QLineEdit::placeholder {
        color: rgba(255,255,255,180);
        font-style: italic;
    }
    QPushButton {
        background-color: #ff6600;
        color: white;
        border: none;
        border-radius: 10px;
        padding: 10px 20px;
        font-family: 'Tektur';
        font-size: 16px;
        font-weight: bold;
    }
    QPushButton:hover { background-color: #ffa31a; }
    QPushButton:pressed { background-color: #cc5200; }
         </string>
        </property>
        <widget class="QLabel" name="labelJoinTitle">
         <property name="geometry">
          <rect>
           <x>150</x>
           <y>40</y>
           <width>200</width>
           <height>40</height>
          </rect>
         </property>
         <property name="text">
          <string>Join Game</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
        <widget class="QLineEdit" name="inputCode">
         <property name="geometry">
          <rect>
           <x>125</x>
           <y>120</y>
           <width>250</width>
           <height>41</height>
          </rect>
         </property>
         <property name="placeholderText">
          <string>Enter room code</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnConfirmJoin">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>200</y>
           <width>180</width>
           <height>45</height>
          </rect>
         </property>
         <property name="text">
          <string>Join</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnBackJoin">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>260</y>
           <width>180</width>
           <height>45</height>
          </rect>
         </property>
         <property name="text">
          <string>Back</string>
         </property>
        </widget>
       </widget>
      </widget>
      <widget class="QWidget" name="page_wait">
       <widget class="QFrame" name="frameWait">
        <property name="geometry">
         <rect>
          <x>370</x>
          <y>90</y>
          <width>500</width>
          <height>541</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">
         QFrame {
             background-color: rgba(0, 0, 0, 0.6);
             border: 2px solid #ff6600;
             border-radius: 20px;
         }
         QLabel {
             color: white;
             font-family: 'Tektur';
             font-size: 20px;
             font-weight: bold;
         }
         QListWidget {
             background-color: rgba(255,255,255,0.1);
             border-radius: 10px;
             color: white;
             font-family: 'Tektur';
             font-size: 16px;
         }
         QPushButton {
             background-color: #ff6600;
             color: white;
             border: none;
             border-radius: 10px;
             padding: 10px 20px;
             font-family: 'Tektur';
             font-size: 16px;
             font-weight: bold;
         }
         QPushButton:hover { background-color: #ffa31a; }
         QPushButton:pressed { background-color: #cc5200; }
         </string>
        </property>
        <widget class="QLabel" name="labelRoomCode">
         <property name="geometry">
          <rect>
           <x>60</x>
           <y>50</y>
           <width>381</width>
           <height>41</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">QLabel {
    background-color: rgba(0, 0, 0, 150);   /* color de fondo */
    color: white;                  /* color del texto */
    border: 2px solid #ff6600    /* borde naranja */
    border-radius: 0px;            /* sin esquinas redondeadas */
    padding: 5px;
    font-weight: bold;
}
</string>
         </property>
         <property name="text">
          <string>ROOM CODE: </string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
        <widget class="QListWidget" name="listPlayers">
         <property name="geometry">
          <rect>
           <x>100</x>
           <y>220</y>
           <width>300</width>
           <height>200</height>
          </rect>
         </property>
        </widget>
        <widget class="QPushButton" name="btnStartGame">
         <property name="geometry">
          <rect>
           <x>70</x>
           <y>450</y>
           <width>180</width>
           <height>45</height>
          </rect>
         </property>
         <property name="text">
          <string>Start Game</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnRefresh">
         <property name="geometry">
          <rect>
           <x>270</x>
           <y>450</y>
           <width>161</width>
           <height>45</height>
          </rect>
         </property>
         <property name="text">
          <string>Refresh </string>
         </property>
        </widget>
        <widget class="QLabel" name="labelLobbyStatus">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>100</y>
           <width>181</width>
           <height>40</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">QLabel {
    background: transparent;
    border: none;
    color: white;   /* o el color que prefieras */
    font-weight: bold;
}
</string>
         </property>
         <property name="text">
          <string>Players 1/8</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
        <widget class="QPushButton" name="btnChooseCar">
         <property name="geometry">
          <rect>
           <x>100</x>
           <y>150</y>
           <width>301</width>
           <height>45</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">#btnChooseCar {
    background-color: #053356; /* azul retro */
    color: #faf2b7;
    font-weight: bold;
    border: 2px solid #1cd3dc; /* borde oscuro para contraste */
    border-radius: 8px;
    padding: 8px 16px;
    font-size: 18px;
}

#btnChooseCar:hover {
	color: #053356;
	font-weight: bold;
    background-color: #00BFFF; /* azul eléctrico */
}

#btnChooseCar:pressed {
    background-color: #137ec2;
}
</string>
         </property>
         <property name="text">
          <string>Choose you car</string>
         </property>
        </widget>
       </widget>
      </widget>
      <widget class="QWidget" name="page_car">
       <widget class="QPushButton" name="btnBackToLobby">
        <property name="geometry">
         <rect>
          <x>510</x>
          <y>520</y>
          <width>171</width>
          <height>51</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">#btnBackToLobby {
    background-color: #053356; /* azul retro */
    color: #faf2b7;
    font-weight: bold;
    border: 2px solid #1cd3dc; /* borde oscuro para contraste */
    border-radius: 8px;
    padding: 8px 16px;
    font-size: 18px;
}

#btnBackToLobby:hover {
	color: #053356;
	font-weight: bold;
    background-color: #00BFFF; /* azul eléctrico */
}

#btnBackToLobby:pressed {
    background-color: #137ec2;
}
</string>
        </property>
        <property name="text">
         <string>Select this car</string>
        </property>
       </widget>
       <widget class="QLabel" name="textSelect">
        <property name="geometry">
         <rect>
          <x>279</x>
          <y>120</y>
          <width>691</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QCommandLinkButton" name="btnNextCar">
        <property name="geometry">
         <rect>
          <x>850</x>
          <y>320</y>
          <width>41</width>
          <height>41</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QCommandLinkButton" name="btnPrevCar">
        <property name="geometry">
         <rect>
          <x>320</x>
          <y>320</y>
          <width>41</width>
          <height>41</height>
         </rect>
        </property>
        <property name="layoutDirection">
         <enum>Qt::RightToLeft</enum>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QLabel" name="labelCreateInfo_2">
        <property name="geometry">
         <rect>
          <x>300</x>
          <y>140</y>
          <width>611</width>
          <height>61</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">QLabel {
    background-color: #222222;     /* gris oscuro de fondo */
    color: white;                  /* texto blanco */
    border: 3px solid #00ffff;             /* cian neón */
    border-radius: 20px;           /* bordes redondeados */
    font-weight: bold;             /* negrita */
    font-size: 20px;
    padding: 6px 20px;             /* margen interno */
    qproperty-alignment: 'AlignCenter';  /* centra el texto */
}

</string>
        </property>
        <property name="text">
         <string>The best exciting moment: Select your car!</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QLabel" name="labelCarImage">
        <property name="geometry">
         <rect>
          <x>360</x>
          <y>210</y>
          <width>491</width>
          <height>301</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">#labelCarImage {
    background-color: rgba(0, 0, 0, 130);
    border: 3px solid #00ffff;             /* cian neón */
    border-radius: 15px;
    padding: 8px;
}

#labelCarName {
    color: #00ffff;
    font-weight: bold;
    font-family: &quot;Tektur SemiBold&quot;;
    background-color: transparent;
    text-align: center;
    font-size: 18px;
}
</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="textFormat">
         <enum>Qt::MarkdownText</enum>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="page_maps">
       <widget class="QListWidget" name="listMaps">
        <property name="geometry">
         <rect>
          <x>350</x>
          <y>170</y>
          <width>581</width>
          <height>441</height>
         </rect>
        </property>
       </widget>
       <widget class="QScrollBar" name="verticalScrollBar">
        <property name="geometry">
         <rect>
          <x>910</x>
          <y>170</y>
          <width>20</width>
          <height>441</height>
         </rect>
        </property>
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
       <widget class="QPushButton" name="btnSelectMaps">
        <property name="geometry">
         <rect>
          <x>530</x>
          <y>630</y>
          <width>231</width>
          <height>51</height>
         </rect>
        </property>
        <property name="text">
         <string>Select and back to menu</string>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
